products/bluebell/core/src/intermediate_representation/emitter.rs

```diff
@@ -3,44 +3,95 @@ use crate::ast::nodes::*;
 use crate::ast::visitor::AstVisitor;
 use crate::constants::{TraversalResult, TreeTraversalMode};
 use crate::intermediate_representation::primitives::*;
+use crate::intermediate_representation::symbol_table::SymbolTable;
 use crate::parser::lexer::SourcePosition;
+use log::info;
 
-use log::warn;
 use std::mem;
 
+/// Byte Code Generation Process
```

Comment:
IMO there should be `products/bluebell/README.md` with high level overview and how to run things, and this explaination of how things work should be in `docs/` like the other stuff? Or at least replicated.

Reply:
Yeah, that is a very sensible suggestion and definitely something that will be incorporated over time. I have added an `README.md` to help getting started, but more detailed documentation on architecture etc. is not within the scope of this PR.